src/main/java/bio/terra/service/snapshot/flight/export/SnapshotExportDumpFirestoreStep.java

```diff
@@ -0,0 +1,105 @@
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+public class SnapshotExportDumpFirestoreStep implements Step {
+  private final SnapshotService snapshotService;
+  private final UUID snapshotId;
+  private final ObjectMapper objectMapper;
+
+  public SnapshotExportDumpFirestoreStep(
+      SnapshotService snapshotService, UUID snapshotId, ObjectMapper objectMapper) {
+    this.snapshotService = snapshotService;
+    this.snapshotId = snapshotId;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    Snapshot snapshot = snapshotService.retrieve(snapshotId);
+    // String firestoreProject = snapshot.getProjectResource().getGoogleProjectId();
+    String firestoreProject =
+        firestoreProject =
+            snapshot
+                .getSnapshotSources()
+                .get(0)
+                .getDataset()
+                .getProjectResource()
+                .getGoogleProjectId();
+
+    try {
+      FirestoreOptions firestoreOptions =
+          FirestoreOptions.getDefaultInstance().toBuilder()
+              .setProjectId(firestoreProject)
+              .setCredentials(GoogleCredentials.getApplicationDefault())
+              .build();
+      String sourceDatasetId = snapshot.getSnapshotSources().get(0).getDataset().getId().toString();
+      String collectionName = String.format("%s-files", sourceDatasetId);
+      final Firestore db = firestoreOptions.getService();
+      final CollectionReference collection = db.collection(collectionName);
+      QuerySnapshot queryResult = collection.get().get();
+      GcsChannelWriter writer = makeWriterForDumpFile(context);
+      queryResult
+          .getDocuments()
+          .forEach(
+              d -> {
+                final Map<String, Object> data = d.getData();
+                try {
+                  String serializedData = objectMapper.writeValueAsString(data);
+                  writer.write(serializedData);
+                } catch (JsonProcessingException e) {
+                  // todo: handle this
+                  e.printStackTrace();
+                } catch (IOException e) {
+                  // todo: handle this
+                  e.printStackTrace();
+                }
+              });
+      writer.close();
```

Comment:
In general you shouldn't need to call `close()` directly since anything `AutoCloseable` should be opened using try-with-resources. In this case something like
```
   try (GcsChannelWriter writer = makeWriterForDumpFile(context)) {
      ...
   }
```